zng/record.go

```diff
@@ -2,26 +2,50 @@ package zng
 
 import (
 	"encoding/json"
-	"errors"
 	"fmt"
-	"strings"
 
 	"github.com/mccanne/zq/zcode"
 )
 
-var ErrColumnMismatch = errors.New("zng record mismatch between columns in type and columns in value")
-
 type TypeRecord struct {
+	Context
+	ID      int
 	Columns []Column
-	Key     string
+	LUT     map[string]int
+	TsCol   int
+	//XXX get rid of Key and use ID as context-unique type id
+	Key string
+}
+
+func CopyTypeRecord(id int, r *TypeRecord) *TypeRecord {
+	return &TypeRecord{
+		Columns: r.Columns,
+		LUT:     r.LUT,
+		TsCol:   r.TsCol,
+		Key:     r.Key,
+	}
 }
 
-func recordString(columns []Column) string {
-	return columnList("record[", columns, "]")
+//XXX this should take ID as arg
+func NewTypeRecord(id int, columns []Column) *TypeRecord {
+	r := &TypeRecord{
+		Columns: columns,
+		TsCol:   -1,
+		Key:     ColumnString("", columns, ""), //XXX
+	}
+	r.createLUT()
+	return r
+}
+
+//XXX
+func RecordString(columns []Column) string {
```

Comment:
Shouldn't this be called `RecordTypeString`?

Reply:
How about` TyperRecordString`